controllers/hcloudmachinetemplate_controller_test.go

```diff
@@ -247,5 +247,59 @@ var _ = Describe("HCloudMachineTemplateReconciler", func() {
 				}, timeout, time.Second).Should(BeTrue())
 			})
 		})
+		Context("HCloudMachineTemplate Webhook Validation", func() {
```

Comment:
the tests here should work with envtest. This means that you create an object and depending on whether you want it to succeed or fail, you expect an error on creation or not. Same with update. For update, you need to first create it, then fetch it, then update it, and expect either an error or not